ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs

```diff
@@ -353,26 +362,23 @@ interested in this candidate at all.
 chainForkSuffix
   :: (HasHeader header, HasHeader block,
       HeaderHash header ~ HeaderHash block)
-  => ChainFragment block  -- ^ Current chain.
-  -> ChainFragment header -- ^ Candidate chain
-  -> Maybe (ChainFragment header)
-chainForkSuffix current candidate =
-    case ChainFragment.intersectChainFragments current candidate of
-      Nothing                         -> Nothing
-      Just (_, _, _, candidateSuffix) -> Just candidateSuffix
+  => AnchoredFragment block  -- ^ Current chain.
+  -> AnchoredFragment header -- ^ Candidate chain
+  -> AnchoredFragment header
+chainForkSuffix current candidate = candidateSuffix
+  where
+    (_, _, candidateSuffix) = AnchoredFragment.intersect current candidate
 
 selectForkSuffixes
   :: (HasHeader header, HasHeader block,
       HeaderHash header ~ HeaderHash block)
-  => ChainFragment block
-  -> [(FetchDecision (ChainFragment header), peerinfo)]
-  -> [(FetchDecision (ChainFragment header), peerinfo)]
+  => AnchoredFragment block
+  -> [(FetchDecision (AnchoredFragment header), peerinfo)]
+  -> [(FetchDecision (AnchoredFragment header), peerinfo)]
 selectForkSuffixes current chains =
     [ (mchain', peer)
     | (mchain,  peer) <- chains
-    , let mchain' = do
-            chain <- mchain
-            chainForkSuffix current chain ?! FetchDeclineChainNoIntersection
+    , let mchain' = chainForkSuffix current <$> mchain
```

Comment:
So we made no intersection impossible, but didn't check it anywhere, just by assumption, but we didn't `assert` or document here what that assumption is. Makes me nervous :-)

Reply:
Fixed in the newest version.